src/transformers/generation/logits_process.py

```diff
@@ -172,11 +172,73 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
 
 class TemperatureLogitsWarper(LogitsWarper):
     r"""
-    [`LogitsWarper`] for temperature (exponential scaling output probability distribution).
+    [`LogitsWarper`] for temperature (exponential scaling output probability distribution) which effectively means that
+    we can control the randomness of the predicted tokens.
+
+    <Tip>
+
+    Make sure that `do_sample=True` is included in the `generate` arguments otherwise the temperature value won't have
+    any effect. As noted in `generation_strategies#Assisted-Decoding`, this parameter along with assisted decoding will
+    help improving the latency.
+
+    </Tip>
 
     Args:
         temperature (`float`):
-            The value used to module the logits distribution.
+            The value used to module the logits distribution. A value of `1.` will make no difference in the output
+            with respect to calling `generate` without temperature.
+
+    Examples:
+
+    ```python
+    >>> import torch
+    >>> from transformers import AutoTokenizer, AutoModelForCausalLM
```

Comment:
Uhmm... I have 2 requests for the example:
1. Let's avoid nested functions, to make the example as plain as possible
2. Let's try to craft an example with 20-30 lines

Let your creative juices go all out here :) A viable alternative could be e.g. to show what happens when you call generate with a different seed, and that a low temperature leads to seed-invariant results

Reply:
Ahh, yeah, I had a kind of voice inside me whispering: *don't put that super-duper function there...*

Let me see what can take out of this poor brain :brain: I have :wink:

Reply:
Finally I couldn't squeeze my neurons to get something beyond your suggestions, hope the pr looks ok now, but don't loose the hope, eventually  we will get there :wink: